its/tester.go

```diff
@@ -0,0 +1,218 @@
+package its
+
+import (
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+
+	"io/ioutil"
+
+	"os"
+
+	"fmt"
+
+	"runtime"
+)
+
+var (
+	regexpCommandLine = regexp.MustCompile("('[^']*')|(\\S+)")
+)
+
+type IntegrationTest interface {
+	Run(description string, action func())
+	Cmd(commandLine string) IntegrationTest
+	DriverName() string
+	ShouldContainLines(count int) IntegrationTest
+	ShouldContainLine(index int, text string) IntegrationTest
+	ShouldEqualLine(index int, text string) IntegrationTest
+	ShouldSucceed() IntegrationTest
+	ShouldSucceedWith(message string) IntegrationTest
+	ShouldFail() IntegrationTest
+	ShouldFailWith(errorMessage string) IntegrationTest
+	TearDown()
+}
+
+func NewIntegrationTest(t *testing.T) IntegrationTest {
+	storagePath, _ := ioutil.TempDir("", "docker")
+
+	return &dockerMachineTest{
+		t:           t,
+		storagePath: storagePath,
+	}
+}
+
+type dockerMachineTest struct {
+	t           *testing.T
+	storagePath string
+
+	Description string
+	RawOutput   string
+	Lines       []string
+	Err         error
+	Failed      bool
+}
+
+func (dmt *dockerMachineTest) Run(description string, action func()) {
+	dmt.Description = description
+	dmt.RawOutput = ""
+	dmt.Lines = nil
+	dmt.Err = nil
+	dmt.Failed = false
+
+	fmt.Print("\033[1;33m[..]\033[0m " + description)
```

Comment:
Why not make some small helper functions which make reading the use of these escape codes a bit more readable? e.g.:

```
func red(msg string) {
    return fmt.Sprintf("\e[31m%s\e[0m")
}
```

That said, the motivation for moving over to Go-based integration tests is largely for running on Windows, no?  Have you run these tests on git bash?  In my recollection the available Windows terminals don't play very well with escape codes, so maybe we should consider skipping them.